modify verification start from snapshot

cryptoseeking opened the pull request:
`replay_push_block` worked when starting from snapshot.

1. replaying block_log , this time included backup can not be found in forkdb.
2. replaying reversible_block , this time included backup can be found in forkdb because backup reversible blocks are put into forkdb

`push_block` verifies a received block which contain a root sibling backup

Commit:
check blog irreversible block's previous backup when start from snapshot

Commit:
repair start from snapshot

Commit:
Merge branch 'develop' of github.com:armoniax/amax.meta.chain into develop

Commit:
1. uniform full_signed_block struct
2. modify start_block logic to intercept exception on backup block production.
3. modify amax-blocklog to print backup block
4. modify net_plugin to adapt new edition full block.

metamaster2021 commented on libraries/chain/fork_database.cpp:
```diff
@@ -678,5 +683,18 @@ namespace eosio
+      {
+         const auto &previdx = my->index.get<best_backup_by_prev>();
+         auto best_backup_state = previdx.lower_bound( head_prev );
+         while( best_backup_state != previdx.end() ){
```
should check (*best_backup_state)->header.previous == head_prev

metamaster2021 commented on libraries/chain/fork_database.cpp:
```diff
@@ -678,5 +683,18 @@ namespace eosio
          return block_state_ptr();
       }
+
+      block_state_ptr  fork_database::get_producer_backup_block( name prod, const block_id_type head_prev ) const
```
rename   head_prev -> previous

metamaster2021 commented on plugins/net_plugin/net_plugin.cpp:
```diff
@@ -2536,10 +2543,11 @@ namespace eosio {
-               full_signed_block_packer::unpack(ds, *main_block, backup_block);
-
-               if (backup_block) {
+               full_signed_block_ptr fptr = std::make_shared<full_signed_block>();
```
use this one:
               signed_block_ptr backup_block;
               full_signed_block::unpack(ds, *main_block, backup_block);

Commit:
optimization:
    1 net work transfer bandwith
    2 block header extension about backup block
    3 code format

Commit:
modify logical

Commit:
update tester

Commit:
change default
